packages/ui-components/Common/ChangeHistory/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { ChevronDownIcon, ClockIcon } from '@heroicons/react/24/outline';
+import classNames from 'classnames';
+import type { FC, ComponentProps } from 'react';
+
+import type { LinkLike } from '#ui/types.js';
+
+import styles from './index.module.css';
+
+export type HistoryChange = {
+  versions: Array<string>;
+  label: string;
+  url?: string;
+};
+
+type ChangeHistoryProps = ComponentProps<'div'> & {
+  label: string;
+  changes: Array<HistoryChange>;
+  as?: LinkLike;
+};
+
+const ChangeHistory: FC<ChangeHistoryProps> = ({
+  label = 'History',
+  changes = [],
+  className,
+  as: As = 'a',
+  'aria-label': ariaLabel = label,
+  ...props
+}) => (
+  <div className={classNames('relative', 'inline-block', className)} {...props}>
+    <details className="group">
+      <summary className={styles.summary} role="button" aria-haspopup="menu">
+        <ClockIcon className="size-4" />
+        <span>{label}</span>
+        <ChevronDownIcon className="size-3 group-open:rotate-180 motion-safe:transition-transform" />
```

Comment:
```suggestion
        {children}
```

Instead of defining the icons and content within the component itself, it might be better to pass them as `children` from the outside. This way, we can reuse it not only in `ChangeHistory` but also in many other places

Reply:
I'm planning to refactor the contents out of this component into a dropdown component in my next PR. I believe we wanted a specific implementation at the moment.

Reply:
Either works. Ultimately the tooling will integrate with whatever is added here